src/cli/common.rs

```diff
@@ -375,88 +376,55 @@ where
     Ok(utils::ExitCode(0))
 }
 
-pub(crate) fn list_targets(distributable: DistributableToolchain<'_>) -> Result<utils::ExitCode> {
-    let mut t = process().stdout().terminal();
-    let manifestation = distributable.get_manifestation()?;
-    let config = manifestation.read_config()?.unwrap_or_default();
-    let manifest = distributable.get_manifest()?;
-    let components = manifest.query_components(distributable.desc(), &config)?;
-    for component in components {
-        if component.component.short_name_in_manifest() == "rust-std" {
-            let target = component
-                .component
-                .target
-                .as_ref()
-                .expect("rust-std should have a target");
-            if component.installed {
-                let _ = t.attr(terminalsource::Attr::Bold);
-                let _ = writeln!(t.lock(), "{target} (installed)");
-                let _ = t.reset();
-            } else if component.available {
-                let _ = writeln!(t.lock(), "{target}");
-            }
-        }
-    }
-
-    Ok(utils::ExitCode(0))
+pub(crate) fn list_targets(
```

Comment:
I've read somewhere that Rust "hates boolean parameters", as the readability would be problematic esp. when you pass in boolean literals.
Your particular use of this parameter here seems fine, but considering future contributions (esp. external ones) I would've refactored this into two functions calling one private impl passing `true` and `false` respectively.
That being said, I'll let you decide on this one.

Reply:
IMO the notion that Rust "hates boolean parameters" is pretty dogmatic/overblown. I agree that `bool` is not a great choice for widely used functions (especially in the public API), or if there are multiple `bool` arguments in the same function call, or if the role of the `bool` value is particularly sensitive (removing some code). In this case, none of those are relevant.

> Your particular use of this parameter (passing in named variables) here seems fine, but considering future contributions (esp. external ones) I would've refactored this into two functions calling one private impl passing `true` and `false` respectively.

To my eyes, this solution is worse than the problem. Function wrappers that add limited functionality to an inner function they call just make names verbose and obscure shared purpose.

As such, I think this usage is fine. In other cases, the right solution IME might be to have a custom enum -- for this case maybe something like `enum LocalState { Installed, NotInstalled }`, variants like `::Yes` or `True` are another anti-pattern, but that has a decent amount of "overhead" in terms of extra code needed so I don't think doing so here is worth it.